build.go

```diff
@@ -68,6 +71,139 @@ func if_not_exists(cmd string, cb func()) {
+	r, err := zip.OpenReader(src)
+    if err != nil {
+        return err
+    }
+    defer func() {
+        if err := r.Close(); err != nil {
+            panic(err)
+        }
+    }()
+
+    os.MkdirAll(dest, 0755)
+
+    // Closure to address file descriptors issue with all the deferred .Close() methods
+    extractAndWriteFile := func(f *zip.File) error {
+       rc, err := f.Open()
+        if err != nil {
+            return err
+        }
+        defer func() {
+            if err := rc.Close(); err != nil {
+                panic(err)
+            }
+        }()
+
+        path := filepath.Join(dest, f.Name)
+
+        if f.FileInfo().IsDir() {
+            os.MkdirAll(path, f.Mode())
+        } else {
+            os.MkdirAll(filepath.Dir(path), f.Mode())
+            f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+            if err != nil {
+                return err
+            }
+            defer func() {
+                if err := f.Close(); err != nil {
+                    panic(err)
+                }
+            }()
+
+            _, err = io.Copy(f, rc)
+            if err != nil {
+                return err
+            }
+        }
+        return nil
+    }
+
+    for _, f := range r.File {
+			if strings.Contains((*f).Name, fileToExtract) {
+        err := extractAndWriteFile(f)
+        if err != nil {
+					return err
+        }
+				return nil
+			}
+    }
+
+    return nil
+}
+
+func downloadFromUrl(url string) string {
+	tokens := strings.Split(url, "/")
+	fileName := filepath.Join(os.Getenv("GOPATH"), tokens[len(tokens)-1])
+	fmt.Println("Downloading", url, "to", fileName)
+
+	output, err := os.Create(fileName)
+	if err != nil {
+		fmt.Println("Error while creating", fileName, "-", err)
+		return ""
+	}
+	defer output.Close()
+
+	response, err := http.Get(url)
+	if err != nil {
+		fmt.Println("Error while downloading", url, "-", err)
+		return ""
+	}
+	defer response.Body.Close()
+
+	n, err := io.Copy(output, response.Body)
+	if err != nil {
+		fmt.Println("Error while downloading", url, "-", err)
+		return ""
+	}
+
+	fmt.Println(n, "bytes downloaded.")
+	return fileName
+}
+
+func join(separator string, args ...string) string {
+	return strings.Join(args, separator)
+}
+
+func installProtoc() {
+		if_not_exists("protoc", func() { // since we require a specific version anyway, wouldn't it be better to include this binary in the initial install filesystem?
+			const GOARCH string = runtime.GOARCH
+			const GOOS string = runtime.GOOS
+			if GOOS == "linux" && GOARCH == "amd64" { // check os and arch. If 64-bit linux, download binary. Other cases should be added as we start supporting other oses/architectures.
+				protocVersionNumber := strings.Split(ProtocVersion, " ")[1]
+				filename := downloadFromUrl(join("", "https://github.com/google/protobuf/releases/download/v", protocVersionNumber, "/protoc-", protocVersionNumber, "-linux-x86_64.zip"))
```

Comment:
In this case, I don't see any real difference here. If google decides to change their legacy repo of this binary to something that no longer works, and does it as a commit, than the govendor method would be superior, but the odds of that seem very similar to the odds of them deleting the legacy binaries all together. So it would reduce the currently low risk to a lower risk. Seems like it would make more sense to just eliminate the risk. But having said that, if I get some time to do it, I will try to remember to change the download method to govendor...but it's pretty low priority in my mind.